pyannote/audio/core/model.py

```diff
@@ -702,3 +705,208 @@ def default_map_location(storage, loc):
             raise e
 
         return model
+
+    def push_to_hub(
+        self,
+        repo_id: str,
+        commit_message: Optional[str] = None,
+        private: Optional[bool] = None,
+        token: Optional[Union[bool, str]] = None,
+        create_pr: bool = False,
+        revision: str = None,
+        commit_description: str = None,
+        tags: Optional[List[str]] = None,
+        **deprecated_kwargs,
+    ) -> None:
+        """
+        Upload the {object_files} to the 🤗 Model Hub.
+
+        Parameters:
+            repo_id (`str`):
+                The name of the repository you want to push your {object} to. It should contain your organization name
+                when pushing to a given organization.
+            commit_message (`str`, *optional*):
+                Message to commit while pushing. Will default to `"Upload {object}"`.
+            private (`bool`, *optional*):
+                Whether or not the repository created should be private.
+            token (`bool` or `str`, *optional*):
+                The token to use as HTTP bearer authorization for remote files. If `True`, will use the token generated
+                when running `huggingface-cli login` (stored in `~/.huggingface`). Will default to `True` if `repo_url`
+                is not specified.
+            create_pr (`bool`, *optional*, defaults to `False`):
+                Whether or not to create a PR with the uploaded files or directly commit.
+            revision (`str`, *optional*):
+                Branch to push the uploaded files to.
+            commit_description (`str`, *optional*):
+                The description of the commit that will be created
+            tags (`List[str]`, *optional*):
+                List of tags to push on the Hub.
+        """
+
+        api = HfApi()
+
+        _ = api.create_repo(
+            repo_id, private=private, token=token, exist_ok=True, repo_type="model"
+        )
+
+        model_type = str(type(self)).split("'")[1].split(".")[-1]
+
+        with TemporaryDirectory() as tmpdir:
```

Comment:
Would it be cleaner to define a `.save_pretrained` method, where we save the config + checkpoint to a specified dir? This would allow users to have local copies of the models.

In `push_to_hub`, we could then simply do:
```python
with TemporaryDirectory() as tmpdir:
    self.save_pretrained(tmpdir)
    return api.upload_folder(
                repo_id=repo_id,
                folder_path=tmpdir,
                ...
    )
```

Reply:
Agree, it would be great indeed!